specs/_features/eip7594/polynomial-commitments-sampling.md

```diff
@@ -678,28 +675,16 @@ def recover_cells_and_kzg_proofs(cell_indices: Sequence[CellIndex],
     # Convert cells to coset evals
     cosets_evals = [cell_to_coset_evals(cell) for cell in cells]
 
-    reconstructed_data = recover_data(cell_indices, cosets_evals)
-
-    for cell_index, coset_evals in zip(cell_indices, cosets_evals):
-        start = cell_index * FIELD_ELEMENTS_PER_CELL
-        end = (cell_index + 1) * FIELD_ELEMENTS_PER_CELL
-        assert reconstructed_data[start:end] == coset_evals
+    polynomial_coeff = recover_polynomial_coeff(cell_indices, cosets_evals)
 
-    recovered_cells = [
-        coset_evals_to_cell(reconstructed_data[i * FIELD_ELEMENTS_PER_CELL:(i + 1) * FIELD_ELEMENTS_PER_CELL])
-        for i in range(CELLS_PER_EXT_BLOB)]
-    
-    polynomial_eval = reconstructed_data[:FIELD_ELEMENTS_PER_BLOB]
-    polynomial_coeff = polynomial_eval_to_coeff(polynomial_eval)
     recovered_proofs = [None] * CELLS_PER_EXT_BLOB
-    for i, cell_index in enumerate(cell_indices):
-        recovered_proofs[cell_index] = bytes_to_kzg_proof(proofs_bytes[i])
+    recovered_cells = [None] * CELLS_PER_EXT_BLOB
```

Comment:
Justin noted that we should do cells = [] and then call append instead of doing it like this -- Not opinionated, will change before moving from Draft